### PR TITLE
FEATURE: EGL-166 support unconditional labels

### PR DESCRIPTION
**Pull request by phong-nguyen-duy**
---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15807/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17306/ : FAILURE in 7m 44s

File: libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/label/LabelSubSection.tsx [hunk 1]
```diff
@@ -4,54 +4,165 @@ import { useIntl } from "react-intl";
...
     const defaultValue =
         properties?.controls?.comparison?.labelConfig?.unconditionalValue ||
-        formatMessage({ id: defaultLabelKey });
+        formatMessage({ id: defaultLabelKeys.nonConditionalKey });
 
+    const conditionalEnabled =
+        !!properties?.controls?.comparison?.labelConfig?.isConditional &&
+        calculationType !== CalculateAs.RATIO;
+    let positiveDefaultValue, negativeDefaultValue, equalsDefaultValue;
+    if (conditionalEnabled) {
```
Comment (khanhhuynh-gooddata) [inline libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/label/LabelSubSection.tsx:76]: consider to remove the condition and change the variable name:
defaultValue -> unconditionalValue
positiveDefaultValue -> positiveValue
negativeDefaultValue -> negativeValue
equalsDefaultValue -> equalsValue

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15812/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17312/ : FAILURE in 7m 45s

**Comment by khanhhuynh-gooddata** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15814/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17313/ : FAILURE in 11m 28s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15817/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17320/ : SUCCESS in 11m 49s

File: libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/label/LabelSubSection.tsx (hunk 1 above)
Comment (phong-nguyen-duy) [reply libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/label/LabelSubSection.tsx]: The condition is necessary, these values should not be computed if users do not enable the conditional labels.
BTW, I renamed as your suggestion. Thank you.

File: libs/sdk-ui-charts/src/interfaces/comparison.ts [hunk 2]
```diff
@@ -122,6 +130,52 @@ export interface ILabelConfig {
      * </table>
      */
     unconditionalValue?: string;
+
+    /**
+     * This property specify the label of the positive comparison value
+     * <br/>
+     * Primary &gt; Secondary
```
[the remark this answers is not in the document]
Comment (phong-nguyen-duy) [reply libs/sdk-ui-charts/src/interfaces/comparison.ts]: done

**Comment by phong-nguyen-duy** [comment]
extended test - backstop

**Comment by khanhhuynh-gooddata** [comment]
extended test - backstop

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15824/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17323/ : SUCCESS in 11m 44s

**Comment by yenkins** [comment]
Link to Storybook test result: https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-backstop-tests/2459/artifact/libs/sdk-ui-tests/backstop/output/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/15825/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-pull-request-dispatcher-pipeline/1403/ : SUCCESS in 27m 21s

**Review by khanhhuynh-gooddata — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4104

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3219/ : SUCCESS in 11m 07s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17328/ : SUCCESS in 11m 42s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/832/ : SUCCESS in 11m 23s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4378/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1099/ : FAILURE in 1m 28s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/739/ : SUCCESS in 8m 39s

**Pull request merged**